sklearn/linear_model/glm.py

```diff
@@ -0,0 +1,872 @@
+        """
+        return np.power(mu, self.power)
+
+    def unit_variance_derivative(self, mu):
+        """The derivative of the unit variance of a Tweedie distribution is
+        v(mu)=power*mu**(power-1).
+        """
+        return self.power*np.power(mu, self.power-1)
+
+    def unit_deviance(self, y, mu):
+        p = self.power
+        if p == 0:
+            #NormalDistribution
+            return (y-mu)**2
+        if p == 1:
+            #PoissonDistribution
+            return 2 * (np.where(y==0,0,y*np.log(y/mu))-y+mu)
+        elif p == 2:
+            #GammaDistribution
+            return 2 * (np.log(mu/y)+y/mu-1)
+        else:
+            #return 2 * (np.maximum(y,0)**(2-p)/((1-p)*(2-p))
+            #    - y*mu**(1-p)/(1-p) + mu**(2-p)/(2-p))
+            return 2 * (np.power(np.maximum(y,0), 2-p)/((1-p)*(2-p))
+                - y*np.power(mu, 1-p)/(1-p) + np.power(mu, 2-p)/(2-p))
+
+    def likelihood(self, y, X, beta, phi, weight=1):
+        raise NotImplementedError('This function is not (yet) implemented.')
+
+
+class NormalDistribution(TweedieDistribution):
+    """Class for the Normal (aka Gaussian) distribution"""
+    def __init__(self):
+        super(NormalDistribution, self).__init__(power=0)
+
+GaussianDistribution = NormalDistribution
+
+class PoissonDistribution(TweedieDistribution):
+    """Class for the scaled Poisson distribution"""
+    def __init__(self):
+        super(PoissonDistribution, self).__init__(power=1)
+
+class GammaDistribution(TweedieDistribution):
+    """Class for the Gamma distribution"""
+    def __init__(self):
+        super(GammaDistribution, self).__init__(power=2)
+
+class InverseGaussianDistribution(TweedieDistribution):
+    """Class for the scaled InverseGaussianDistribution distribution"""
+    def __init__(self):
+        super(InverseGaussianDistribution, self).__init__(power=3)
+
+class GeneralizedHyperbolicSecand(ExponentialDispersionModel):
+    """A class for the von Generalized Hyperbolic Secand (GHS) distribution.
+
+    The GHS distribution is for data y in (-inf, inf).
+    """
+    def __init__(self):
+        self._lower_bound = -np.Inf
+        self._upper_bound = np.Inf
+
+    @property
+    def lower_bound(self):
+        return self._lower_bound
+
+    @property
+    def upper_bound(self):
+        return self._upper_bound
+
+    def in_y_range(self, x):
+        np.logical_and(
+            np.greater(x, self.lower_bound),
+            np.less(x, self.lower_bound)
+            )
+
+    def unit_variance(self, mu):
+        return 1 + mu**2
+
+    def unit_variance_derivative(self, mu):
+        return 2*mu
+
+    def unit_deviance(self, y, mu):
+        return (2*y*(np.arctan(y) - np.arctan(mu))
+            + np.log((1+mu**2)/(1+y**2)))
+
+
+
+class GeneralizedLinearModel(LinearModel, RegressorMixin):
+    """
+    Class to fit a Generalized Linear Model (GLM) based on reproductive
+    Exponential Dispersion Models (EDM).
+
+    Assumptions:
```

Comment:
Most of the below belongs in the [User Guide](https://github.com/scikit-learn/scikit-learn/blob/master/doc/modules/linear_model.rst)